examples/serving/README.md

```diff
@@ -0,0 +1,95 @@
+<!--
+  Copyright (c) Microsoft Corporation
+  All rights reserved.
+
+  MIT License
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+  documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+  the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+  to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+  The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+  BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+  DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+-->
+
+
+# Model Serving on PAI
+
+This guide introduces how to run model serving workload on PAI.
+Serving system for machine learning models is designed for production environments, which makes it easy to deploy new algorithms and experiments to users.
+The following contents show some basic model serving examples, other customized serving code can be run similarly.
+
+
+## Contents
+
+1. [Basic environment](#basic-environment)
+2. [Serving a TensorFlow model](#serving-a-tensorflow-model)
+
+
+## Basic environment
+
+First of all, PAI runs all jobs in Docker container.
+
+[Install Docker-CE](https://docs.docker.com/install/linux/docker-ce/ubuntu/) if you haven't. Register an account at public Docker registry [Docker Hub](https://hub.docker.com/) if you do not have a private Docker registry.
+
+We use TensorFlow model serving as an example, for how to serve a TensorFlow model, please refer to its [serving tutorial](https://www.tensorflow.org/serving/serving_basic).
+
+You can also jump to [Serving a TensorFlow model](#serving-a-tensorflow-model) using [pre-built images](https://hub.docker.com/r/paiexample/pai.example.tensorflow-serving/) on Docker Hub.
+
+We need to build a TensorFlow serving image with GPU support to serve a TensorFlow model on PAI. We use the [TensorFlow serving Dockerfile](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/tools/docker/Dockerfile.devel-gpu) provided in its tutorial.
+
+Build the TensorFlow serving Docker image for PAI. Download the Dockerfile and built it directly.
+
+```bash
+$ wget -O Dockerfile.example.tensorflow-serving https://raw.githubusercontent.com/tensorflow/serving/master/tensorflow_serving/tools/docker/Dockerfile.devel-gpu
+$ sudo docker build -f Dockerfile.example.tensorflow-serving .
```

Comment:
this dockerfile doesn't include hadoop binary, so the job cannot run if user directly use this image.

Reply:
the data and model are prepared locally in this example

Reply:
updated